src/Compilers/Core/Portable/Syntax/SyntaxList.cs

```diff
@@ -21,9 +23,9 @@ public override string Language
             }
         }
 
-        protected override SyntaxTree SyntaxTreeCore => this.Parent.SyntaxTree;
+        protected override SyntaxTree SyntaxTreeCore => this.Parent!.SyntaxTree;
```

Comment:
⚠️  This appears to violate the documented behavior of the base property (emphasis mine):

> Returns SyntaxTree that owns the node **or null if node does not belong to a SyntaxTree**

```suggestion
        protected override SyntaxTree SyntaxTreeCore => this.Parent?.SyntaxTree;
```

Reply:
It does violate the documented behavior but it is also the behavior. Safe bug to fix here.